pysmiles/smiles_helper.py

```diff
@@ -579,3 +578,78 @@ def increment_bond_orders(molecule, max_bond_order=3):
         molecule.edges[idx, jdx]['order'] = new_order
         missing_bonds[idx] -= edge_missing
         missing_bonds[jdx] -= edge_missing
+
+def mark_chiral_atoms(molecule):
+    """
+    For all nodes tagged as chiral, figure out the three
+    substiuents and annotate the node with a tuple that
+    has the order in which to rotate. This essentially
+    corresponds to the definition of an improper dihedral
+    angle centered on the chiral atom.
+    """
+    chiral_nodes = nx.get_node_attributes(molecule, 'stereo')
+    for node, direction in chiral_nodes.items():
+        # discard the node with the lowest index because we
+        # are looking down that node; it is not part of the
+        # stereo projection
+        neighbors = sorted(molecule.neighbors(node))[1:]
+        if len(neighbors) != 3:
+            n = len(neighbors) + 1
+            msg = (f"Chiral node {node} has {n} neighbors, which "
+                    "is different than the four expected for "
+                    "tetrahedral chirality. If the chiral center "
+                    "contains hydrogen you need to set explicit_hydrogen "
+                    "to True.")
+            raise ValueError(msg)
+        # the default is anti-clockwise sorting indicated by '@'
+        # in this case the nodes are sorted with increasing
+        # node index; however @@ means clockwise and the
+        # order of nodes is reversed (i.e. with decreasing index)
+        if direction == '@@':
+            neighbors = [neighbors[0], neighbors[2], neighbors[1]]
```

Comment:
Why 0, 2, 1 and not 2, 1, 0?

Reply:
x, 2, 1, 0 is the same improper dihderal angle as x, 0, 1, 2. To change the direction you must permute only the last two atoms in the definition. But I need to double check this as well

Reply:
I've double checked this